src/main/java/com/passpnu/passwordmanager/service/PasswordEntityServiceImpl.java

```diff
@@ -171,4 +212,65 @@ public void deletePassword(Long serviceId, AuthUserDetailsDto userDto) {
         Long userId = userDto.getId();
         passwordRepository.deleteByUserIdAndServiceId(userId, serviceId);
     }
+
+    @Override
+    public List<AnalysisDto> analyzeSystem(AuthUserDetailsDto authUser) throws EncryptionException, NameNotFoundException {
+        final String encryptionKey = authUser.getEncryptionKey();
+
+        List<PasswordEntity> entities = passwordRepository.findAllByUserId(authUser.getId());
+
+        List<AnalysisDto> analysisDtoList = entities.stream()
+                .map(entity -> {
+                    try {
+                        return customPasswordMapper.entityToAnalysisDto(entity, passwordEncryptor, encryptionKey);
+                    } catch (EncryptionException e) {
+                        throw new UncheckedEncryptionException(e.getMessage());
+                    }
+                })
+                .toList();
+
+        final int listSize = analysisDtoList.size();
```

Comment:
Guava has a method for having a cartesian product of sets (https://guava.dev/releases/snapshot/api/docs/com/google/common/collect/Sets.html#cartesianProduct(java.util.List))

Reply:
I don't actually need cartesian product. It will increase complexity twice.